app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java

```diff
@@ -0,0 +1,125 @@
+package com.invisibleteam.goinvisible.mvvm.edition;
+
+
+import android.app.Activity;
+import android.content.IntentSender;
+import android.location.LocationManager;
+import android.util.Log;
+
+import com.google.android.gms.common.api.PendingResult;
+import com.google.android.gms.common.api.Status;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResult;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+
+import javax.annotation.Nullable;
+
+class GpsEstablisher {
+    private static final String TAG = GpsEstablisher.class.getSimpleName();
+
+    private static final int GPS_REQUEST_CODE_DEFAULT = 1234;
+    private static final long LOCATION_REQUEST_INTERVAL = 100;
+    private static final long LOCATION_REQUEST_FASTEST_INTERVAL = 100;
+
+    private GoogleLocationApiEstablisher googleLocationApiEstablisher;
+    private LocationManager locationManager;
+    private Activity activity;
```

Comment:
do you really need activity in this place? If yes, better will be weak reference notation